of_design_doc.md

```diff
@@ -92,14 +96,16 @@ ExpandedMatrixType = @of(
     halved=of(Array, n/2, n),
 )
 
-ConcreteExpanded = ExpandedMatrixType(; n=10)
-# This creates:
-# - original: 10×10 matrix
-# - padded: 11×11 matrix
-# - doubled: 20×10 matrix  
-# - halved: 5×10 matrix  (n/2 must result in an integer, error if not)
+# Create instance - all non-constant fields default to zero
+instance = ExpandedMatrixType(; n=10)
+# This creates an instance with:
+# - original: 10×10 zero matrix
+# - padded: 11×11 zero matrix
+# - doubled: 20×10 zero matrix  
+# - halved: 5×10 zero matrix  (n/2 must result in an integer, error if not)
 
-rand(ConcreteExpanded)
+# Generate random instance
+rand(ExpandedMatrixType; n=10)
```

Comment:
```suggestion
rand(of(ExpandedMatrixType; n=10))
```

Reply:
@sunxd3 For clarity, let's make sure to remove these methods

```julia
zero(ExpandedMatrixType; n=10)
rand(ExpandedMatrixType; n=10)
flat = latten(MatrixType, instance; rows=3, cols=4)
unflatten(MatrixType, flat; rows=3, cols=4)
```

in favour of

```julia
zero(of(ExpandedMatrixType; n=10))
rand(of(ExpandedMatrixType; n=10))
flat = latten(of(MatrixType, instance; rows=3, cols=4))
unflatten(of(MatrixType, flat; rows=3, cols=4))
```

for the principle that there should be only one way of doing things.

Reply:
done now